src/modules/commander/HealthAndArmingChecks/checks/loggerCheck.hpp

```diff
@@ -0,0 +1,58 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2024 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#pragma once
+
+#include "../Common.hpp"
+
+#include <uORB/Subscription.hpp>
+#include <uORB/SubscriptionMultiArray.hpp>
+#include <uORB/topics/logger_status.h>
+
+class LoggerChecks : public HealthAndArmingCheckBase
+{
+public:
+	LoggerChecks() = default;
+	~LoggerChecks() = default;
+
+	void checkAndReport(const Context &context, Report &reporter) override;
+
+private:
+	uORB::SubscriptionMultiArray<logger_status_s> _logger_status_sub{ORB_ID::logger_status};
+
+#if defined(CONFIG_MODULES_LOGGER)
+	DEFINE_PARAMETERS_CUSTOM_PARENT(HealthAndArmingCheckBase,
+					(ParamInt<px4::params::SDLOG_MODE>) _param_sdlog_mode
```

Comment:
```suggestion
					(ParamInt<px4::params::SDLOG_MODE>) _param_sdlog_mode
```

Using `SDLOG_MODE` here makes logger a hard dependency for commander.


Options
1. use param_find/param_get and tolerate the potential for SDLOG_MODE to not actually exist
2. a new commander param to configure if active healthy logging is a requirement or not for arming

Reply:
This kind of reminds me of the need for multi-stage arming.
 - arm stage 1 (aka pre-arm, but the name is currently overloaded)
 - modules like logger get a chance to start and report their status
   - another example would be ESCs given a chance to spool up and report in
 - if everything reports in then commander fully arms and the system is allowed to takeoff, etc

Reply:
> Using SDLOG_MODE here makes logger a hard dependency for commander.

That's why I did this. 
```
#if defined(CONFIG_MODULES_LOGGER)
```
> This kind of reminds me of the need for multi-stage arming.

Yeah it leads towards that. The intent of this PR tho is just to be able to report the status of the logger so that we know when it's logging or not.

Reply:
> a new commander param to configure if active healthy logging is a requirement or not for arming

Probably not worth doing that given the wide array of conditions in SDLOG_MODE.

![image](https://github.com/PX4/PX4-Autopilot/assets/37091262/91193835-b4d0-4acc-a05f-6001e1ca98f9)